### PR TITLE
Backport for PHP 5.6

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: tests/CommonChunksTestOutputBuilder.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
```
Comment (keradus) [inline tests/CommonChunksTestOutputBuilder.php:1]: why new file for a fork ?

File: tests/DifferTest.php [hunk 2]
```diff
@@ -475,19 +475,21 @@ public function textForNoNonDiffLinesProvider(): array
...
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessageRegExp #^"to" must be an array or string\.$#
+     */
     public function testDiffInvalidToType()
     {
-        $this->expectException('\InvalidArgumentException');
-        $this->expectExceptionMessageRegExp('#^"to" must be an array or string\.$#');
```
Comment (keradus) [inline tests/DifferTest.php]: I think it's available in phpunit5, isn't it?
then, we could still use it as we do php 5.6+

File: tests/CommonChunksTestOutputBuilder.php (hunk 1 above)
Comment (SpacePossum) [reply tests/CommonChunksTestOutputBuilder.php:1]: this is an anonymous class in the original repo, which doesn't play well on PHP < 7

File: tests/DifferTest.php [hunk 3]
```diff
@@ -498,27 +500,17 @@ public function testDiffInvalidToType()
...
-            {
-                return '';
-            }
-
-            public function getChunks(array $diff, $lineThreshold)
-            {
-                return $this->getCommonChunks($diff, $lineThreshold);
-            }
-        };
+        $output = new CommonChunksTestOutputBuilder();
```
Comment (keradus) [inline tests/DifferTest.php:510]: here is that new class used.
I would not refactor external, forked code here - only do what's needed to downgrade php

File: tests/DifferTest.php (hunk 2 above)
Comment (SpacePossum) [reply tests/DifferTest.php]: ah damn, I started out with support for 5.5.9 when I didn't release we only need 5.6, will revert :+1:

File: tests/DifferTestTest.php [hunk 4]
```diff
@@ -39,20 +39,19 @@ public function provideDiffWithLineNumbers()
...
         require_once __DIR__ . '/DifferTest.php';
         $test  = new DifferTest();
-        $tests = $test->provideDiffWithLineNumbers();
-
-        $tests = \array_filter(
-            $tests,
-            function ($key) {
-                return !\is_string($key) || false === \strpos($key, 'non_patch_compat');
-            },
-            ARRAY_FILTER_USE_KEY
```
Comment (keradus) [inline tests/DifferTestTest.php]: wasn't it passing on 5.6 ?

File: tests/autoload.php [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<?php
+
+require_once __DIR__.'/../vendor/autoload.php';
+require_once __DIR__.'/CommonChunksTestOutputBuilder.php';
```
Comment (keradus) [inline tests/autoload.php]: without extra class, file is not needed

File: tests/DifferTestTest.php (hunk 4 above)
Comment (SpacePossum) [reply tests/DifferTestTest.php]: same as https://github.com/PHP-CS-Fixer/diff/pull/1#discussion_r134253039, will revert ;)

**Comment by keradus** [comment]
:+1: for package rename, we sadly need it

Please do set up travis and packagist

-----

> README.md

extra minimal, claiming to not use it :D

> composer.json

I would extend `description` to mention it's only php-downgraded fork

> ChangeLog.md

not needed, let us have same release numbers as original package (yet, we don't need to provide all of them)

**Comment by SpacePossum** [comment]
Not sure if the anonymous class can be handled better, I don't want to add two classes into one file to prevent the extra files.
I think this is close to fine as is.

File: tests/CommonChunksTestOutputBuilder.php (hunk 1 above)
Comment (keradus) [reply tests/CommonChunksTestOutputBuilder.php:1]: ok, cool :+1:

File: phpunit.xml.dist [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/6.2/phpunit.xsd"
-         bootstrap="vendor/autoload.php"
+         bootstrap="tests/autoload.php"
```
Comment (keradus) [inline phpunit.xml.dist]: add `autoload-dev` to `composer.json` and no need for manual autoload

Comment (SpacePossum) [reply phpunit.xml.dist]: awesome, didn't know about it :)

Comment (keradus) [reply phpunit.xml.dist]: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.4/composer.json#L61
;P

Comment (SpacePossum) [reply phpunit.xml.dist]: done :)

**Comment by SpacePossum** [comment]
merged

**Comment by keradus** [comment]
great work !

**Pull request closed without merging**